sys/phydat/phydat_str.c

```diff
@@ -78,14 +80,35 @@ const char *phydat_unit_to_str(uint8_t unit)
     }
 }
 
-char phydat_scale_to_str(int8_t scale)
+char phydat_scale_to_str(int8_t scale, uint8_t unit)
 {
+    /* extra handling for multi and non dimensional units */
+    switch (unit) {
+        case UNIT_M2:       /* rescale for 2d unit */
+            if (scale % 2 != 0) {
```

Comment:
returning `\0` will result in a different handling in [phydat_dump](https://github.com/RIOT-OS/RIOT/blob/master/sys/phydat/phydat_str.c#L39), i.e., for instance E notation. So output is not broken.